### PR TITLE
OPA - enable and test

### LINKED ISSUE #2970: Add unit tests for OPA
nice to have

### LINKED ISSUE #3001: Final enablement of OPA in the UI
This issue should only be worked on, if everything else is done and OPA should finally be visible in the dashboard.

Following files need to be touched. List is under construction. 
- [ ] settings/admin/template.html: in line 224 add 
```
<km-tab label="Constraint Templates">
  <km-constraint-templates-list></km-constraint-templates-list>
</km-tab>
```
- [ ] wizard/step/cluster/template.html: un-comment code in line 161
- [ ] wizard/step/summary/template.html: un-comment code in line 104
- [ ] cluster/cluster-details/edit-cluster/template.html: un-comment code in line 94
- [ ] cluster/cluster-details/template.html: in line 242 add
```
<km-tab label="Constraints"
        *ngIf="isOPAEnabled()">
  <km-constraint-list [constraints]="constraints"
                      [cluster]="cluster"
                      [projectID]="projectID"
                      [isClusterRunning]="isClusterRunning"></km-constraint-list>
</km-tab>
```

### PR DESCRIPTION
**Pull request by kgroschoff**
**What this PR does / why we need it**:
Enable all components that are related to OPA and add unit tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2970
Fixes #3001

**Special notes for your reviewer**:
/

**Release note**:

```release-note
Added support for OPA to UI
```

### CODE REVIEW

**Comment by kgroschoff** [comment]
/test pre-dashboard-test-e2e

/hold cancel

@maciaszczykm / @floreks PTAL

File: src/app/shared/entity/opa.ts [hunk 1]
```diff
@@ -50,77 +67,77 @@ export class CRD {
...
-  shortNames: string[];
+  shortNames?: string[];
 }
 
 export class Validation {
-  openAPIV3Schema: JSONSchemaProps;
+  openAPIV3Schema?: JSONSchemaProps;
 }
 
 export class JSONSchemaProps {
```
Comment (floreks) [inline src/app/shared/entity/opa.ts:82]: Are those structures used anywhere or will they be used in any near future?

Comment (kgroschoff) [reply src/app/shared/entity/opa.ts:82]: You can specify the `Validation` fields during constraint template creation, but other than that I'm not sure if there is some feature coming up (same for other structs that are related to constraints & constraint templates). 
Do you think we should only specify e.g. `object` for everything that isn't "needed" atm?

Comment (floreks) [reply src/app/shared/entity/opa.ts:82]: Ye. It's hard to track then which structures are actually used by us. If it is not used in the code, I'd simply remove it and use `object`.

Comment (kgroschoff) [reply src/app/shared/entity/opa.ts:82]: okay, will change it :)

Comment (kgroschoff) [reply src/app/shared/entity/opa.ts]: @floreks done

**Pull request merged**